packages/structures/src/channels/Channel.ts

```diff
@@ -0,0 +1,203 @@
+import { DiscordSnowflake } from '@sapphire/snowflake';
+import type {
+	TextChannelType,
+	ThreadChannelType,
+	APIChannel,
+	APIPartialChannel,
+	ChannelType,
+	GuildChannelType,
+	GuildTextChannelType,
+	ChannelFlags,
+} from 'discord-api-types/v10';
+import { Structure } from '../Structure.js';
+import { ChannelFlagsBitField } from '../bitfields/ChannelFlagsBitField.js';
+import { kData } from '../utils/symbols.js';
+import type { Partialize } from '../utils/types.js';
+import type { ChannelPermissionMixin } from './mixins/ChannelPermissionMixin.js';
+import type { ChannelWebhookMixin } from './mixins/ChannelWebhookMixin.js';
+import type { DMChannelMixin } from './mixins/DMChannelMixin.js';
+import type { GuildChannelMixin } from './mixins/GuildChannelMixin.js';
+import type { TextChannelMixin } from './mixins/TextChannelMixin.js';
+import type { ThreadChannelMixin } from './mixins/ThreadChannelMixin.js';
+import type { ThreadOnlyChannelMixin } from './mixins/ThreadOnlyChannelMixin.js';
+import type { VoiceChannelMixin } from './mixins/VoiceChannelMixin.js';
+
+export type PartialChannel = Channel<ChannelType, Exclude<keyof APIChannel, keyof APIPartialChannel>>;
+
+/**
+ * The data stored by a {@link Channel} structure based on its {@link (Channel:class)."type"} property.
+ */
+export type ChannelDataType<Type extends ChannelType | 'unknown'> = Type extends ChannelType
+	? Extract<APIChannel, { type: Type }>
+	: APIPartialChannel;
+
+/**
+ * Represents any channel on Discord.
+ *
+ * @typeParam Type - Specify the type of the channel being constructed for more accurate data types
+ * @typeParam Omitted - Specify the properties that will not be stored in the raw data field as a union, implement via `DataTemplate`
+ * @remarks Although this class _can_ be instantiated directly for any channel type,
+ * it's intended to be subclassed with the appropriate mixins for each channel type.
+ */
+export class Channel<
+	Type extends ChannelType | 'unknown' = ChannelType,
+	Omitted extends keyof ChannelDataType<Type> | '' = '',
+> extends Structure<ChannelDataType<Type>, Omitted> {
+	/**
+	 * The template used for removing data from the raw data stored for each Channel.
+	 *
+	 * @remarks This template is only guaranteed to apply to channels constructed directly via `new Channel()`.
+	 * Use the appropriate subclass template to remove data from that channel type.
+	 */
+	public static override DataTemplate: Partial<APIChannel> = {};
```

Comment:
Ngl, DataTemplate should be a symbol

Reply:
What's the motivation for that? This is meant to be used by the end-user to allow them to remove unwanted properties from the structure

At first glance making it a symbol only seems to make it more obscure and potentially harder to use

Reply:
The point is this isn't for users to just _define_ unless they know what they're doing... I mean this ties into https://github.com/discordjs/discord.js/pull/10900#discussion_r2181068448

Reply:
It's meant to be `readonly` (and was in many places, but forgotten in others). Does that satisfy your worry about users defining it?

Reply:
Not really, imo still should be a symbol but 🤷